cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/index.js

```diff
@@ -229,7 +187,14 @@ export default class ColumnsTab extends Component {
       );
     }
 
-    let displayHeaders = this.state.headers;
+    let index = -1;
+    let displayHeaders = this.state.headers.map(header => {
```

Comment:
Its not the same index. This index calculation skips the `isDetail` rows that gets added to headers list when user clicks on any row with row info.